frontend/containers/forms/uploader/component.tsx

```diff
@@ -0,0 +1,183 @@
+                message: error?.message || (
+                  <FormattedMessage
+                    defaultMessage="Something went wrong with the file upload"
+                    id="qpDURb"
+                  />
+                ),
+              });
+              setIsUploading(false);
+            }
+          }
+        })
+      );
+
+      onUpload(uploadedFiles);
+      setIsUploading(false);
+      setFilesLength(0);
+    },
+    [maxSize, onUpload, setError, setMaxSizeError]
+  );
+
+  const onDropAccepted = async (files: File[]) => {
+    clearErrors('project_images_attributes');
+    uploadFiles(files);
+  };
+
+  const onDropRejected = async (fileRejections: FileRejection[]) => {
+    setIsUploading(false);
+    fileRejections.forEach((rejection) => {
+      rejection.errors.forEach((error) => {
+        switch (error.code) {
+          case 'file-too-large':
+            setMaxSizeError(rejection.file);
+            break;
+          default:
+            setError('project_images_attributes' as Path<FormValues>, { message: error.message });
+        }
+      });
+    });
+  };
+
+  const { getRootProps, getInputProps } = useDropzone({
+    accept: fileTypes,
+    disabled: disabled || !!isUploading,
+    maxSize,
+    maxFiles,
+    multiple: maxFiles > 1,
+    onDropAccepted,
+    onDropRejected,
+  });
+
+  useEffect(() => {
+    if (isUploading) clearErrors('project_images_attributes');
+  }, [clearErrors, isUploading]);
+
+  return (
+    <div className="h-full">
+      <div
+        className={classNames(
+          'h-full rounded-md border-2 border-dashed  border-beige py-10 px-4 text-center text-sm focus:outline-green-700',
+          {
+            'cursor-pointer': !disabled,
+            'opacity-50': disabled,
+          }
+        )}
+        {...getRootProps()}
+      >
+        <input {...getInputProps()} />
+        <input
+          className="hidden"
+          name="project_images_attributes"
+          onProgress={console.log}
+          {...register('project_images_attributes' as Path<FormValues>)}
+        />
+        <Image src="/images/upload-gallery.svg" width="38" height="39" alt="Upload file" />
+        {!isUploading ? (
+          <div className="mt-6">
+            <p className="text-sm font-normal text-gray-800">
+              <FormattedMessage
+                defaultMessage="<a>Browse</a> or drag and drop"
+                id="hpOE0K"
+                values={{
+                  a: (chunks: string) => (
+                    <span className="font-medium text-green-dark">{chunks}</span>
+                  ),
+                }}
+              />
+            </p>
+            <p className="text-gray-600">
+              <FormattedMessage defaultMessage="PNG, JPG up to 5MB" id="BHrBCl" />
+            </p>
+          </div>
+        ) : (
+          <p>
+            <FormattedMessage
+              defaultMessage="Uploading {l} files..."
```

Comment:
The `{l}` param will be shown to translators. It would be better to have a self explaining name:
```suggestion
              defaultMessage="Uploading {filesCount} files..."
```